src/main/java/github/scarsz/discordsrv/objects/managers/JdbcAccountLinkManager.java

```diff
@@ -201,16 +255,99 @@ private void dropExpiredCodes() {
+                statement.setLong(2, System.currentTimeMillis());
+                try (final ResultSet result = statement.executeQuery()) {
+                    if (result.next()) {
+                        uuidString = result.getString("uuid");
+                        if (jedisPool != null) {
+                            final long expiration = result.getLong("expiration");
+                            final String codeKey = "linking/code/" + code;
+                            final String uuidKey = "linking/uuid/" + uuidString;
+                            try (Jedis jedis = jedisPool.getResource()) {
+                                jedis.set(codeKey, uuidString);
+                                jedis.expireAt(codeKey, expiration);
+                                jedis.set(uuidKey, code);
+                                jedis.expireAt(uuidKey, expiration);
+                            }
+                        }
+                    }
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        if (uuidString == null) return null;
+        return UUID.fromString(uuidString);
+    }
+
+    public String getLinkingCode(UUID uuid) {
+        final String uuidString = uuid.toString();
+        String code = null;
+        if (jedisPool != null) {
+            try (Jedis jedis = jedisPool.getResource()) {
+                code = jedis.get("linking/uuid/" + uuidString);
+            }
+        }
+        if (code == null) {
+            try (final PreparedStatement statement = connection.prepareStatement("select `code`, `expiration` from " + codesTable + " where `uuid` = ? and `expiration` >= ?")) {
+                statement.setString(1, uuidString);
+                statement.setLong(2, System.currentTimeMillis());
+                try (final ResultSet result = statement.executeQuery()) {
+                    if (result.next()) {
+                        code = result.getString("code");
+                        if (jedisPool != null) {
+                            final long expiration = result.getLong("expiration");
+                            final String codeKey = "linking/code/" + code;
+                            final String uuidKey = "linking/uuid/" + uuidString;
+                            try (Jedis jedis = jedisPool.getResource()) {
+                                jedis.set(codeKey, uuidString);
+                                jedis.expireAt(codeKey, expiration);
+                                jedis.set(uuidKey, code);
+                                jedis.expireAt(uuidKey, expiration);
+                            }
+                        }
+                    }
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return code;
+    }
+
     @Override
     public Map<String, UUID> getLinkingCodes() {
-        dropExpiredCodes();
```

Comment:
Removal would cause codes not to be removed other than when the class in constructed

Reply:
I am using
`keyExpiryScheduler.schedule(() -> dropCode(uuid, finalCode), 5, TimeUnit.MINUTES);`
to remove the keys. The cleanup on construction is there just in case of improper shutdown, or shut down prior to the scheduled drop.